### PR TITLE
Drop php 5.6 support

### LINKED ISSUE #63: Consider moving to PHP 7
- According to [PHP's Roadmap](http://php.net/supported-versions.php), PHP 5.6 will be actively supported only up to 1st January 2017.
- Until Manalize is spread _all around the world_, PHP 7 will be the norm.
- Manalize is a tool used by developers, on their own machines. It should not be an issue, as any respectful developer should have the latest PHP version installed. :p 
- We're depriving ourself from using some really cool features (see https://github.com/manala/manalize/pull/62#issuecomment-256407154 for instance)
- Really ? :trollface:

### PR DESCRIPTION
**Pull request by chalasr**
Closes #63

### CODE REVIEW

File: src/Command/Setup.php [hunk 1]
```diff
@@ -75,7 +75,7 @@ protected function execute(InputInterface $input, OutputInterface $output)
         return 0;
     }
 
-    private function setupDependencies(SymfonyStyle $io, MetadataBag $metadata)
+    private function setupDependencies(SymfonyStyle $io, MetadataBag $metadata) : \Generator
```
Comment (ogizanagi) [inline src/Command/Setup.php]: There is no convention yet about this AFAIK, but I personally prefer:

``` diff
- private function setupDependencies(SymfonyStyle $io, MetadataBag $metadata) : \Generator
+ private function setupDependencies(SymfonyStyle $io, MetadataBag $metadata): \Generator
```

(no space between the closing parentheses and the colon)

WDYT ?

Comment (chalasr) [reply src/Command/Setup.php]: I kept the space from what I can see in some popular repositories, but I'm looking for a standard... Do you have such?

Comment (ogizanagi) [reply src/Command/Setup.php]: I based my choice on https://wiki.php.net/rfc/scalar_type_hints
But as I said, I don't know any consensus.

Comment (ogizanagi) [reply src/Command/Setup.php]: See https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/2207/files#diff-88b99bb28683bd5b7e3a204826ead112R523

Comment (chalasr) [reply src/Command/Setup.php]: Quick win! 👍

Comment (ogizanagi) [reply src/Command/Setup.php]: This fixer is not released yet however 😢  It's only for 2.0.

Comment (chalasr) [reply src/Command/Setup.php]: I thought about moving to 2.x-dev within this PR. Wdyt?

Comment (ogizanagi) [reply src/Command/Setup.php]: 👍

**Comment by nervo** [comment]
Hm, we still have, and still support some projects in 5.4, 5.5 and 5.6

**Comment by ogizanagi** [comment]
@nervo : It's not about dropping the support of PHP 5.6 in virtualized environments, but on the host on which is used the manalize tool in order to init a project.

**Comment by nervo** [comment]
Did'nt we see that osx (even sierra release) is still in 5.6 ?

**Comment by ogizanagi** [comment]
That's entirely true. However, see https://github.com/manala/manalize/issues/63 :)

**Comment by nervo** [comment]
Hm, ok.
:half_convinced: :)

**Comment by ogizanagi** [comment]
Let's fight tomorrow about that ☕️ 🚬 :D

**Comment by ogizanagi** [comment]
As I did it for formageddon, I added 1ed26b5 to switch to php-cs-fixer v2 ;)

**Comment by chalasr** [comment]
Great, thank you @ogizanagi!

@nervo I understand your pov, I personally had some doubts about this, thinking to people like frontend developers who just do not want to hear about PHP.

But we already have some requirements (which are not pre-installed on osx/linux), so PHP 7 would just be another one, and probably the easier to install

```
$ curl -s https://php-osx.liip.ch/install.sh | bash -s 7.0
```

(if we prefer homebrew, we just have to publish a shell script executing [the 5 commands it needs](https://developerjack.com/blog/2015/12/11/Installing-PHP7-with-homebrew/))

Imho, it is worth it.

**Comment by ogizanagi** [comment]
Let's give @nervo the responsibility for merging (_or closing_) this one (symbolism, U know :trollface: )

**Comment by ogizanagi** [comment]
Thanks @chalasr ! :)

**Pull request merged**